### ISSUE TITLE
Doubts about the Bailando model

### ISSUE THREAD

**Issue by WJ-Fifth** [lisiyao21/Bailando#42]
You have completed a very good model!

I also achieved very good results when I was working on your model. But there are still some questions that are not very clear. Are you experiencing gradient explosion when implementing the Actor-Critic Learning module? My model still converged at the first epoch, and it did have some improvement compared to GPT. However, during the subsequent iterations, L_AC increased significantly and could not continue to converge. And the visualization results also became very strange.

Looking forward for your reply!

**Comment by lisiyao21** [comment]
Deeply sorry for such a late reply. Indeed, the actor critic in this version is an 'on-policy' one. That is, it updates the policy based on the sampled trajectories of current policy weights. Since there is no supervision with the ground truth data anymore, if it goes to long, the policy will gradually 'forget' what it learns from the data and purely focus on the RL reward. The reward, however, do not include many items to maintain its motion quality (it only focuses on beat alignment). That is why it goes bad when finetuning more iterations.

Hope it clarifies.

Best

**Comment by XUCHUANXINA** [comment]
Hi,I want to ask about the paper "Bailando". I used the preprocessed data provided by the author and the project code on GitHub to reproduce the paper. All hyperparameters were set as per the original author's configuration. I conducted the reproduction on PyTorch 2.0.0, Python 3.8, CUDA 11.8, and an RTX 3090. However, the reproduction results differ significantly from those of the author. What could be the reasons for this? Could you offer some assistance? Thank you.
